gimie/make_graph.py

```diff
@@ -0,0 +1,75 @@
+from rdflib import Graph, URIRef, parser
+import hashlib
+from calamus import fields
+from calamus.schema import JsonLDSchema
+
+schema = fields.Namespace("http://schema.org/")
+
+
+def generate_fair_uri(
+    repository_path: str = "https://example.com/Repository/",
+):
+    """given a repository_path, returns a URI with a hash for uniqueness, or the repository URL if it's online"""
+    # Compute the SHA-256 hash of the repository name
+    hash = hashlib.sha256(repository_path.encode()).hexdigest()
+    # TODO https://www.github.com/repoX should return the same result as github.com/repoX
+    if (
+        False
+    ):  # repository_path.is_online(): #we need a method that checks whether the repopath is online or a foldername
+        fair_uri = repository_path
+    else:
+        # Return the FAIR URI in the form "fair:sha256-HASH, truncated to 5 characters to promote readability"
+        fair_uri = (
+            f"gimie:{repository_path}/" + hash[:5]
+        )  # TODO decide on URI we want to use for non online repos
+    return fair_uri
+
+
+class Repository:
+    """A class to represent a GIT repository
+
+    Parameters
+    ----------
+    path: str
+        The path to the git repository.
+
+    Attributes
+    ----------
+    license_url (conform SPDX)
+    programming_language
+    """
+
+    def __init__(self, path: str):
+        self._id = generate_fair_uri(path)
+        self.license_url = "http://example.com/books/1"  # LicenseMetadata(self.path).get_licenses() TODO replace
+        self.programming_language = "python"  # FilesMetadata(self.path).get_programming_lang() TODO replace
+
+
+class RepositorySchema(JsonLDSchema):
+    """A Class which indicates how the attributes of Repository map to schema.org properties conform Calamus"""
+
+    _id = fields.Id()
+    license_url = fields.String(schema.license)
+    programming_language = fields.String(schema.programmingLanguage)
+
+    class Meta:
+        """a metaclass which indicates how the repository should be rdf:typed conform Calamus"""
+
+        rdf_type = schema.SoftwareApplication
+        model = Repository
+
+
+def to_graph(path: str, output_format: str = "ttl"):
```

Comment:
It should either have a more explicit name (_what_ to graph), or be a method inside the Repository class.

If implemented as  a method, I think this could work:

```python
class Repository:
     # [...] Everything you already defined
    
    def to_graph(self, format: str = 'ttl') -> str:
        jsonld_dict = RepositorySchema().dump(self)
        g = Graph()
        g.parse(format="json-ld", data=jsonld_dict)
        g.bind(
            "schema", schema
        )
        return g.serialize(format=format)
```
A few changes in my suggestions:
* returning a string instead of printing it, which makes it easier for the user to do whatever they want with it (can be stored in a variable by doing `graph_string = Repository(url).to_graph()`
* use `self` to point at the software instance itself
* `output_format` -> `format` since this is the only format the user is aware of

Reply:
Agree with your points, reason I had initially used output_format instead of format is because format is a reserved word for the format() method. However, it does not actually cause a problem so might as well make it 'format'.